src/Polly.Core/README.md

```diff
@@ -1,71 +1,68 @@
 
 - `ISyncPolicy`
 - `IAsyncPolicy`
 - `ISyncPolicy<T>`
 - `IAsyncPolicy<T>`
 
-``` csharp
+```csharp
 public abstract class ResiliencePipeline
 {
     public void Execute(Action callback);
 
     public TResult Execute<TResult>(Func<TResult> callback);
-
+    
     public Task ExecuteAsync(Func<CancellationToken, Task> callback, CancellationToken cancellationToken = default);
-
+    
     public Task<TResult> ExecuteAsync(Func<CancellationToken, Task<TResult>> callback, CancellationToken cancellationToken = default);
-
-    public ValueTask ExecuteValueTaskAsync(Func<CancellationToken, ValueTask> callback, CancellationToken cancellationToken = default);
-
-    public ValueTask<TResult> ExecuteValueTaskAsync(Func<CancellationToken, ValueTask<TResult>> callback, CancellationToken cancellationToken = default);
     
-    // omitted for simplicity
+    public ValueTask ExecuteAsync(Func<CancellationToken, ValueTask> callback, CancellationToken cancellationToken = default);
+    
+    public ValueTask<TResult> ExecuteAsync(Func<CancellationToken, ValueTask<TResult>> callback, CancellationToken cancellationToken = default);
+    
+    // Other methods are omitted for simplicity
 }
 ```
 
-The [ResilienceContext](ResilienceContext.cs) is defined as:
+The [`ResilienceContext`](ResilienceContext.cs) is defined as follows:
 
-``` csharp
+```csharp
 public sealed class ResilienceContext
 {
-    public CancellationToken CancellationToken { get; set; }
-
-    public bool IsSynchronous { get; }
-
-    public bool IsVoid { get; }
-
+    public string? OperationKey { get; }
+    public CancellationToken CancellationToken { get; }
     public bool ContinueOnCapturedContext { get; }
-
-    public Type ResultType { get; }
-
-    // omitted for simplicity
+    public ResilienceProperties Properties { get; }
 }
 ```
 
-The `ResiliencePipeline` unifies the 4 different policies used now in Polly. User actions are executed under a single API. The are many methods
-exposed on this class that cover different scenarios:
+The `ResiliencePipeline` class unifies the four different policies currently in use by Polly, enabling user actions to be executed via a single API. This class offers various methods to handle different scenarios:
+
+- Synchronous methods without a return value.
+- Synchronous methods that return a value.
+- Asynchronous methods without a return value.
+- Asynchronous methods that return a value.
+
+> [Note]
+> Polly also provides a `ResiliencePipeline<T>` class. This specialized pipeline is useful for scenarios where the consumer is concerned with only a single type of result.
 
-- Synchronous void methods.
-- Synchronous methods with result.
-- Asynchronous void methods.
-- Asynchronous methods with result.
+### Example: Synchronous `Execute` Method
 
-For example, the synchronous `Execute` method is implemented as:
+For instance, the synchronous `Execute` method is implemented as follows:
```

Comment:
Do we need to explain this implementation detail to consumers?

Reply:
Nope, let's drop this section we don't need that much detail.